python/cudf/cudf/io/parquet.py

```diff
@@ -407,13 +307,29 @@ def read_parquet(
     num_rows=None,
     strings_to_categorical=False,
     use_pandas_metadata=True,
-    use_python_file_object=False,
     categorical_partitions=True,
+    open_options=None,
     *args,
     **kwargs,
 ):
     """{docstring}"""
 
+    # Since use_python_file_object=False is not longer
+    # supported, throw a deprecation warning if the
+    # parameter is used
+    use_python_file_object = kwargs.pop("use_python_file_object", None)
+    if use_python_file_object is not None:
+        py_obj_msg = (
+            "The use_python_file_object argument is now "
+            "deprecated in cudf.read_parquet. "
+            "use_python_file_object=True is now the "
+            "required behavior."
+        )
+        if not use_python_file_object:
+            raise ValueError(py_obj_msg)
+        else:
+            warnings.warn(py_obj_msg, FutureWarning)
```

Comment:
I don't think anyone is explicitly setting `use_python_file_object`.  However, there is no longer any good reason to use `use_python_file_object=False`.  Therefore, I am warning the user if they pass this argument and throwing an error if they ask for `use_python_file_object=False`. I guess this is technically a "breaking" change. However, I doubt there is anyone out there actually *using* this argument.

Reply:
If we want to avoid making this a breaking change, would it make sense to just raise the `FutureWarning` in any case where `use_python_file_object` is specified?

Reply:
Sorry for missing this comment @charlesbluca !

Are you suggesting that we warn, rather than raise a `ValueError` when the user specifies `use_python_file_object=False`?  If so, that was my original plan.  However, I felt pretty strongly that it would do more harm than good to continue supporting the optimized code path for `use_python_file=False`.

With that said, I now realize I could still allow `use_python_file=False`, but fall back on the original behavior (where the entire remote file is read into host memory to avoid using a PyArrow python file object).  Does that sound reasonable?

Reply:
Okay - I decided to only throw the `ValueError` when the user specifies **both** `open_file_options` **and** `use_python_file_object=False`.  This way, there is no breaking change.  However, in the (very) off chance that the user was explicitly defining `use_python_file_object=False` before this change, their code will now be slower.  The good new is that I sincerely doubt that any users will specify `use_python_file_object` at all :)